internal/app/machined/pkg/controllers/k8s/kubelet_spec.go

```diff
@@ -191,49 +195,80 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
 	}
 }
 
-func newKubeletConfiguration(clusterDNS []string, dnsDomain string) *kubeletconfig.KubeletConfiguration {
-	return &kubeletconfig.KubeletConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "kubelet.config.k8s.io/v1beta1",
-			Kind:       "KubeletConfiguration",
-		},
-		StaticPodPath:      constants.ManifestsDirectory,
-		Address:            "0.0.0.0",
-		Port:               constants.KubeletPort,
-		OOMScoreAdj:        pointer.ToInt32(constants.KubeletOOMScoreAdj),
-		RotateCertificates: true,
-		Authentication: kubeletconfig.KubeletAuthentication{
-			X509: kubeletconfig.KubeletX509Authentication{
-				ClientCAFile: constants.KubernetesCACert,
-			},
-			Webhook: kubeletconfig.KubeletWebhookAuthentication{
-				Enabled: pointer.ToBool(true),
-			},
-			Anonymous: kubeletconfig.KubeletAnonymousAuthentication{
-				Enabled: pointer.ToBool(false),
-			},
-		},
-		Authorization: kubeletconfig.KubeletAuthorization{
-			Mode: kubeletconfig.KubeletAuthorizationModeWebhook,
+// NewKubeletConfiguration builds kubelet configuration with defaults and overrides from extraConfig.
+func NewKubeletConfiguration(clusterDNS []string, dnsDomain string, extraConfig map[string]interface{}) (*kubeletconfig.KubeletConfiguration, error) {
+	// check for fields that can't be overridden via extraConfig
+	var multiErr *multierror.Error
+
+	for field, message := range map[string]string{
+		"staticPodPath":  "staticPodPath can't be overridden",
+		"authentication": "authentication can't be overridden",
+		"authorization":  "authorization can't be overridden",
+		"cgroupRoot":     "cgroup configuration can't be overridden",
+		"systemCgroups":  "cgroup configuration can't be overridden",
+		"kubeletCgroups": "cgroup configuration can't be overridden",
+		"port":           "port can't be overridden",
+		"apiVersion":     "apiVersion can't be overridden",
+		"kind":           "kind can't be overridden",
```

Comment:
Should these be documented?

Reply:
yes, they should.

this is a work in progress PR, it's not finished yet